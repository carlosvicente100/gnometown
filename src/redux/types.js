// gnomes
export const fetchRequested = 'FETCH_REQUESTED';
export const fetchRequestedGnomes = 'FETCH_REQUESTED_GNOMES';
export const fetchPendingGnomes = 'FETCH_PENDING_GNOMES';
export const fetchSucceededGnomes = 'FETCH_SUCCEEDED_GNOMES';
export const fetchFailedGnomes = 'FETCH_FAILED_GNOMES';

//filters
export const setAgeFromTo = 'SET_BREWED_FROM_TO';
export const setAgeFrom = 'SET_BREWED_FROM';
export const setAgeTo = 'SET_BREWED_TO';
