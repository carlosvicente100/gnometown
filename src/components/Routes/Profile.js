import React from 'react'
import { useParams } from '@reach/router'
import { useSelector } from 'react-redux'
import Banner from '../Banner/Banner'
import GnomeDetail from '../Gnome/GnomeDetail'

const Profile = () => {
  let { id } = useParams()
  id = parseInt(id)
  const { gnomes = [] } = useSelector((state) => state.gnomesReducer)
  const gnome = gnomes.find((gnome) => gnome.id === id)
  const friends = gnomes.filter(({ name } = u) => gnome.friends.includes(name))

  return (
    <div data-testid="Profile-page">
      <Banner />
      {gnome && <GnomeDetail {...gnome} friends={friends} />}
    </div>
  )
}

export default Profile
