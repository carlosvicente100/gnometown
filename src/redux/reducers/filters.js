import { fetchSetJobList, fetchSetName, fetchResetFilters } from '../types'

export const initialState = {
  jobList: [],
  name: ''
}

export default function reducer(state = initialState, action) {
  switch (action.type) {
    case fetchSetJobList:
      return {
        ...state,
        jobList: action.data.jobs
      }
    case fetchSetName:
      return {
        ...state,
        name: action.data.name
      }
    case fetchResetFilters:
      return {
        ...initialState
      }
    default:
      return {
        ...state
      }
  }
}
