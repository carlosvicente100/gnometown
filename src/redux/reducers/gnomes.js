import { fetchPendingGnomes, fetchSucceededGnomes, fetchFailedGnomes } from '../types'

export const initialState = {
  pending: false,
  error: null,
  townName: '',
  gnomes: []
}

export default function reducer(state = initialState, action) {
  switch (action.type) {
    case fetchPendingGnomes:
      return {
        ...state,
        pending: true
      }
    case fetchSucceededGnomes:
      const townName = Object.keys(action.data)[0]
      return {
        ...state,
        pending: false,
        townName,
        gnomes: action.data[townName]
      }
    case fetchFailedGnomes:
      return {
        ...state,
        pending: false,
        error: true
      }
    default:
      return {
        ...state
      }
  }
}
