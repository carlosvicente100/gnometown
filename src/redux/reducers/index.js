import { combineReducers } from 'redux';
 import gnomesReducer from './gnomes';
import filtersReducer from './filters';

export default combineReducers({
  gnomesReducer,
  filtersReducer,
});
