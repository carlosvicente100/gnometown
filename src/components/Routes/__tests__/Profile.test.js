import React from 'react'
import { Provider } from 'react-redux'
import { createStore } from 'redux'
import { render as rtlRender } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import 'jest-styled-components'

import { Router } from '@reach/router'
import { createHistory, createMemorySource, LocationProvider } from '@reach/router'

import { initialState as initialStateGnomes } from '../../../redux/reducers/gnomes'
import reducer from '../../../redux/reducers'

import Profile from '../Profile'

import GnomeFixtureList from '../../../fixtures/gnomes'

export function renderRouteRedux(
  ui,
  {
    route = '/',
    history = createHistory(createMemorySource(route)),
    initialState,
    store = createStore(reducer, initialState),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    return (
      <Provider store={store}>
        <LocationProvider history={history}>{children}</LocationProvider>
      </Provider>
    )
  }
  return render(ui, { wrapper: Wrapper, ...renderOptions })
}

const initialReducerState = {
  gnomesReducer: initialStateGnomes
}

function render(ui, { initialState, store = createStore(reducer, initialState), ...renderOptions } = {}) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions })
}

describe('Proflie Route Component', () => {
  it('Proflie Route Snapshot', () => {
    expect(
      renderRouteRedux(
        <Router>
          <Profile path="/profile/:id" />
        </Router>,
        {
          route: '/profile/0',
          initialState: { gnomesReducer: { gnomes: GnomeFixtureList[0]['Brastlewark'] } }
        }
      ).baseElement
    ).toMatchSnapshot()
  })
})
