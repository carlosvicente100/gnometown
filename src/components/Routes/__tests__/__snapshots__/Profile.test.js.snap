// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Proflie Route Component Proflie Route Snapshot 1`] = `
.c0 {
  position: relative;
  background: white;
  width: 100%;
  height: 72px;
  display: grid;
  grid-template-columns: 15% 35% 35% 15%;
  grid-template-areas: '. m-town-name m-town-name m-filter';
  margin-bottom: 20px;
}

.c0 label {
  text-align: center;
  grid-area: m-town-name;
  font-family: 'Helvetica';
  font-weight: 300;
  font-size: 34px;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  justify-self: center;
}

.c1 {
  font-size: 20px;
  margin: 20px;
}

.c2 {
  background-color: #7f8fa6;
  font-family: 'Helvetica';
  color: black;
  border-radius: 25px;
  overflow: hidden;
  height: 100%;
  display: grid;
  -webkit-column-gap: 10px;
  column-gap: 10px;
  text-align: center;
  grid-row-gap: 20px;
  grid-template-areas: '.image .' '. title .' '. properties .' '. jobs .' 'friends friends friends';
  padding: 20px;
  margin: auto;
}

.c2 img {
  grid-area: image;
}

.c2 ul,
.c2 h3 {
  margin: 2px;
}

.c2 h2 {
  height: auto;
  grid-area: title;
  font-size: 16px;
  text-transform: uppercase;
}

.c2 .gnome-professions {
  margin: 20px 0px;
  grid-area: jobs;
  font-size: 14px;
}

.c2 .gnome-professions ul {
  list-style: none;
  text-align: left;
  width: 20%;
  margin: auto;
  text-align: left;
}

.c2 .gnome-professions ul li {
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
}

.c2 .gnome-friends {
  grid-area: friends;
  text-align: center;
  font-size: 14px;
}

.c2 .gnome-friends .GnomeList {
  margin: 0px;
}

.c2 .properties {
  grid-area: properties;
  font-size: 14px;
}

.c2 .properties p {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
}

.c2 .properties span {
  font-weight: bold;
}

.c3 {
  height: 200px;
  margin: auto;
  width: 200px;
  border-radius: 100px;
}

.c4 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin: auto;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
}

.c4 .dummy {
  margin: auto;
  margin-left: 5px;
  width: 20px;
  height: 20px;
  border-radius: 50%;
  border: 2px solid black;
}

<body>
  <div>
    <div
      style="outline: none;"
      tabindex="-1"
    >
      <div
        data-testid="Profile-page"
      >
        <div
          class="c0"
        >
          <div
            class="c1"
            data-testid="banner-go-home"
          >
            <a
              href="/"
            >
              <i
                class="fas fa-home"
              />
            </a>
          </div>
          <label
            data-testid="banner-town-name"
          >
            undefined citizens
          </label>
        </div>
        <div
          class="c2"
        >
          <img
            class="c3"
            data-testid="detail-image"
            src="http://www.publicdomainpictures.net/pictures/10000/nahled/thinking-monkey-11282237747K8xB.jpg"
          />
          <h2>
            Tobus Quickwhistle
          </h2>
          <div
            class="properties"
            data-testid="detail-properties"
          >
            <p>
              <span>
                Age: 
              </span>
              306 years
            </p>
            <p>
              <span>
                Height:
              </span>
              107.76 cm
            </p>
            <p>
              <span>
                Weight:
              </span>
               39.07 kg
            </p>
            <div
              class="c4"
            >
              <p>
                Hair Color: 
              </p>
              <div
                class="dummy"
                style="background-color: Pink;"
              />
            </div>
          </div>
          <div
            class="gnome-professions"
            data-testid="detail-professions"
          >
            <h3>
              Professions
            </h3>
            <ul>
              <li>
                Metalworker
              </li>
              <li>
                Woodcarver
              </li>
              <li>
                Stonecarver
              </li>
              <li>
                 Tinker
              </li>
              <li>
                Tailor
              </li>
              <li>
                Potter
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
